doc/source/whatsnew/v0.20.0.txt

```diff
@@ -193,6 +193,7 @@ Map on Index types now return other Index types
 Other API Changes
 ^^^^^^^^^^^^^^^^^
 
+- ``pd.read_csv()`` will now issue a ``UserWarning`` whenever there are conflicting values provided by the ``dialect`` parameter and the user (:issue:`14898`)
```

Comment:
Actually, to be consistent with what we do with `converter` and `dtype` conflicting, I should also issue a `ParserWarning`.  Good catch.